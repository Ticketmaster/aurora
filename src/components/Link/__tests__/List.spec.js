import React from "react";
import { render, Simulate } from "react-testing-library";

import LinkList from "../List";
import LinkListItem from "../ListItem";

const mockFn = jest.fn();

const countries = [
  {
    value: "en-US",
    text: "USA (English)",
    label: "United States"
  },
  {
    value: "en-CA",
    text: "Canada (English)",
    label: "Canada"
  },
  {
    value: "fr-CA",
    text: "Canada (Fran√ßais)",
    label: "Canada"
  }
];

describe("<LinkList />", () => {
  it("renders LinkList correctly when its open", () => {
    const { container } = render(
      <LinkList onItemClick={mockFn} selectedIndex={0}>
        {countries.map((option, index) => (
          <LinkListItem index={index} key={option.value}>
            {option.text}
          </LinkListItem>
        ))}
      </LinkList>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders LinkList correctly with list items as links", () => {
    const { container } = render(
      <LinkList onItemClick={mockFn} selectedIndex={0}>
        {countries.map((option, index) => (
          <LinkListItem
            index={index}
            key={option.value}
            onClick={mockFn}
            href="/"
          >
            {option.text}
          </LinkListItem>
        ))}
      </LinkList>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("onItemClick in LinkList is called when a list item is clicked", () => {
    const onOptionSelected = mockFn;
    const { container } = render(
      <LinkList onItemClick={onOptionSelected} selectedIndex={0}>
        {countries.map((option, index) => (
          <LinkListItem index={index} key={option.value}>
            {option.text}
          </LinkListItem>
        ))}
      </LinkList>
    );

    Simulate.click(container.querySelector(".links__list__item"));
    expect(onOptionSelected).toHaveBeenCalled();
  });

  it("renders content after container", () => {
    const { container } = render(
      <LinkList
        onItemClick={mockFn}
        selectedIndex={0}
        renderAfter={<span>after content</span>}
      >
        {countries.map((option, index) => (
          <LinkListItem
            index={index}
            key={option.value}
            onClick={mockFn}
            href="/"
          >
            {option.text}
          </LinkListItem>
        ))}
      </LinkList>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
