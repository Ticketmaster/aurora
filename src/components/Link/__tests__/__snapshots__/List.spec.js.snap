// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkList /> renders LinkList correctly when its open 1`] = `
<div
  class="sc-bdVaJa links links__list bhMvGM"
>
  <div
    class="sc-bwzfXH bQTlgY"
  >
    <div
      class="links__list__item--selected sc-bxivhb links__list__item gLeqeE"
    >
      USA (English)
    </div>
    <div
      class="sc-bxivhb links__list__item gLeqeE"
    >
      Canada (English)
    </div>
    <div
      class="sc-bxivhb links__list__item gLeqeE"
    >
      Canada (Français)
    </div>
  </div>
</div>
`;

exports[`<LinkList /> renders LinkList correctly with list items as links 1`] = `
<div
  class="sc-bdVaJa links links__list bhMvGM"
>
  <div
    class="sc-bwzfXH bQTlgY"
  >
    <a
      class="sc-EHOje links__list__item hibrNn"
      href="/"
    >
      USA (English)
    </a>
    <a
      class="sc-EHOje links__list__item hibrNn"
      href="/"
    >
      Canada (English)
    </a>
    <a
      class="sc-EHOje links__list__item hibrNn"
      href="/"
    >
      Canada (Français)
    </a>
  </div>
</div>
`;
